test/parallel/test-child-process-disconnect.js

```diff
@@ -108,4 +109,9 @@ if (process.argv[2] === 'child') {
     assert.strictEqual(childFlag, false);
     assert.strictEqual(parentFlag, false);
   });
+
+  process.on('close', function() {
```

Comment:
Perhaps we should be wrapping these event handlers (including `'exit'` and maybe even `'message'`) with a `mustCall()` to ensure the assertions happen?

Reply:
They have a mustCall a bit further up the file.

https://github.com/nodejs/node/pull/19439/files/5b7e0210329095678cd72caf7aa2c51b996ea10a#diff-144ffebe38c5a56b3c2e93c9aeaee664L81

Reply:
That's for the parent process. I think there should be equivalent checks in the child process as well.

Reply:
Right! I also meant to check the child processes close signal there...

Reply:
Turns out ``common.mustCall(...)`` does not work on ``process.on('exit', ...)``, since ``mustCall's`` implementation uses ``process.on('exit', ...)`` itself. 

Luckily we have ``child.on('close', ...)`` now, which is guaranteed to be called only after all messages have been received...

Check the updated commit :)